src/gui_common/IconProgressBar.cs

```diff
@@ -0,0 +1,55 @@
+using Godot;
+
+public class IconProgressBar : ColorRect
+{
+    public bool Disabled { get; private set; }
+    public int Location { get; private set; }
+    public int ActualLocation { get; private set; }
+
+    public void SetBarName(string name)
+    {
+        Name = name;
+    }
+
+    public void SetBarColour(Color colour)
+    {
+        Color = colour;
+    }
+
+    public void SetBarLeftShift(float leftShift)
+    {
+        MarginLeft = leftShift;
+    }
+
+    public void SetBarSize(Vector2 size)
+    {
+        RectSize = size;
+        GetChild<TextureRect>(0).RectSize = new Vector2(size.y, size.y);
+        GetChild<TextureRect>(0).Visible = RectSize.x >= GetChild<TextureRect>(0).RectSize.x;
```

Comment:
Add a comment or separate a function for what this does (adjusting the icon size).